src/commands/commit/apikey.ts

```diff
@@ -0,0 +1,42 @@
+import axios from 'axios'
+
+export class ApiKeyValidator {
+  public apiKey: string | undefined
+  public datadogSite: string
+
+  private isValid?: boolean
+
+  constructor(apiKey: string | undefined, datadogSite: string) {
+    this.apiKey = apiKey
+    this.datadogSite = datadogSite
+  }
+
+  public async isApiKeyValid(): Promise<boolean | undefined> {
+    if (this.isValid === undefined) {
+      this.isValid = await this.validateApiKey()
+    }
+
+    return this.isValid!
+  }
+
+  private getApiKeyValidationURL(): string {
+    return `https://api.${this.datadogSite}/api/v1/validate`
+  }
+
+  private async validateApiKey(): Promise<boolean> {
+    try {
+      const response = await axios.get(this.getApiKeyValidationURL(), {
+        headers: {
+          'DD-API-KEY': this.apiKey,
+        },
+      })
```

Comment:
I think this should be in `api.ts` with other calls to the API (also it avoids duplicating the API base URL crafting)

Reply:
Good point, URL crafting was a bit scattered around. There's now only one source of truth inside `api.ts`.